src/WorkerIndexingHelper.cs

```diff
@@ -42,25 +42,29 @@ private static void AddOutputBinding(BindingInformation outputBinding)
             }
         }
 
-        internal static List<FunctionInformation> IndexFunctions(string baseDir, out List<ErrorRecord> errors)
+        internal static List<FunctionInformation> IndexFunctions(DirectoryInfo baseDir, out List<ErrorRecord> errors)
         {
             errorRecords = new List<ErrorRecord>();
-            if (!Directory.Exists(baseDir))
-            {
-                throw new DirectoryNotFoundException();
-            }
-            if (ContainsLegacyFunctions(Directory.CreateDirectory(baseDir)))
+            if (ContainsLegacyFunctions(baseDir))
             {
                 throw new Exception(AzPowerShellSdkStrings.HybridModelDetected);
             }
-            List<FileInfo> powerShellFiles = GetPowerShellFiles(new DirectoryInfo(baseDir));
+            List<FileInfo> powerShellFiles = GetPowerShellFiles(baseDir);
             List<FunctionInformation> rpcFunctionMetadatas = new List<FunctionInformation>();
 
             foreach (FileInfo powerShellFile in powerShellFiles)
             {
                 rpcFunctionMetadatas.AddRange(IndexFunctionsInFile(powerShellFile));
             }
 
+            IEnumerable<FunctionInformation> noBindingFunctions = rpcFunctionMetadatas.Where(x => x.Bindings.Count() == 0);
+            if (noBindingFunctions.Any())
```

Comment:
If we do not have test cases for this code path, could you please add some tests? Thanks.

Reply:
Done